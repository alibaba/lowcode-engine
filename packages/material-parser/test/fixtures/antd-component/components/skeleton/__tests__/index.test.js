import React from 'react';
import { mount } from 'enzyme';
import Skeleton from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Skeleton', () => {
  const genSkeleton = props =>
    mount(
      <Skeleton loading {...props}>
        Bamboo
      </Skeleton>,
    );
  const genSkeletonButton = props => mount(<Skeleton.Button {...props} />);
  const genSkeletonAvatar = props => mount(<Skeleton.Avatar {...props} />);
  const genSkeletonInput = props => mount(<Skeleton.Input {...props} />);

  mountTest(Skeleton);
  rtlTest(Skeleton);

  it('should without avatar and paragraph', () => {
    const wrapperSmall = genSkeleton({ avatar: false, paragraph: false });
    expect(wrapperSmall.render()).toMatchSnapshot();
  });

  it('should square avatar', () => {
    const wrapperSmall = genSkeleton({ avatar: true, paragraph: false });
    expect(wrapperSmall.render()).toMatchSnapshot();
  });

  describe('avatar', () => {
    it('size', () => {
      const wrapperSmall = genSkeleton({ avatar: { size: 'small' } });
      expect(wrapperSmall.render()).toMatchSnapshot();
      const wrapperDefault = genSkeleton({ avatar: { size: 'default' } });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperLarge = genSkeleton({ avatar: { size: 'large' } });
      expect(wrapperLarge.render()).toMatchSnapshot();
      const wrapperNumber = genSkeleton({ avatar: { size: 20 } });
      expect(wrapperNumber.render()).toMatchSnapshot();
    });

    it('shape', () => {
      const wrapperCircle = genSkeleton({ avatar: { shape: 'circle' } });
      expect(wrapperCircle.render()).toMatchSnapshot();
      const wrapperSquare = genSkeleton({ avatar: { shape: 'square' } });
      expect(wrapperSquare.render()).toMatchSnapshot();
    });
  });

  describe('title', () => {
    it('width', () => {
      const wrapper = genSkeleton({ title: { width: '93%' } });
      expect(wrapper.render()).toMatchSnapshot();
    });
  });

  describe('paragraph', () => {
    it('rows', () => {
      const wrapper = genSkeleton({ paragraph: { rows: 5 } });
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('width', () => {
      const wrapperPure = genSkeleton({ paragraph: { width: '93%' } });
      expect(wrapperPure.render()).toMatchSnapshot();
      const wrapperList = genSkeleton({ paragraph: { width: ['28%', '93%'] } });
      expect(wrapperList.render()).toMatchSnapshot();
    });
  });

  describe('button element', () => {
    it('active', () => {
      const wrapper = genSkeletonButton({ active: true });
      expect(wrapper.render()).toMatchSnapshot();
    });
    it('size', () => {
      const wrapperDefault = genSkeletonButton({ size: 'default' });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperLarge = genSkeletonButton({ size: 'large' });
      expect(wrapperLarge.render()).toMatchSnapshot();
      const wrapperSmall = genSkeletonButton({ size: 'small' });
      expect(wrapperSmall.render()).toMatchSnapshot();
    });
    it('shape', () => {
      const wrapperDefault = genSkeletonButton({ shape: 'default' });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperRound = genSkeletonButton({ shape: 'round' });
      expect(wrapperRound.render()).toMatchSnapshot();
      const wrapperCircle = genSkeletonButton({ shape: 'circle' });
      expect(wrapperCircle.render()).toMatchSnapshot();
    });
  });

  describe('avatar element', () => {
    it('active', () => {
      const wrapper = genSkeletonAvatar({ active: true });
      expect(wrapper.render()).toMatchSnapshot();
    });
    it('size', () => {
      const wrapperSmall = genSkeletonAvatar({ size: 'small' });
      expect(wrapperSmall.render()).toMatchSnapshot();
      const wrapperDefault = genSkeletonAvatar({ size: 'default' });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperLarge = genSkeletonAvatar({ size: 'large' });
      expect(wrapperLarge.render()).toMatchSnapshot();
      const wrapperNumber = genSkeletonAvatar({ size: 20 });
      expect(wrapperNumber.render()).toMatchSnapshot();
    });

    it('shape', () => {
      const wrapperCircle = genSkeletonAvatar({ shape: 'circle' });
      expect(wrapperCircle.render()).toMatchSnapshot();
      const wrapperSquare = genSkeletonAvatar({ shape: 'square' });
      expect(wrapperSquare.render()).toMatchSnapshot();
    });
  });

  describe('input element', () => {
    it('active', () => {
      const wrapper = genSkeletonInput({ active: true });
      expect(wrapper.render()).toMatchSnapshot();
    });
    it('size', () => {
      const wrapperSmall = genSkeletonInput({ size: 'small' });
      expect(wrapperSmall.render()).toMatchSnapshot();
      const wrapperDefault = genSkeletonInput({ size: 'default' });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperLarge = genSkeletonInput({ size: 'large' });
      expect(wrapperLarge.render()).toMatchSnapshot();
    });
  });
});
