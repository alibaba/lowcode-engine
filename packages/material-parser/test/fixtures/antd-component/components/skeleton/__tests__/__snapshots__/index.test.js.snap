// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton avatar element active 1`] = `
<div
  class="ant-skeleton ant-skeleton-element ant-skeleton-active"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-circle"
  />
</div>
`;

exports[`Skeleton avatar element shape 1`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-circle"
  />
</div>
`;

exports[`Skeleton avatar element shape 2`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-square"
  />
</div>
`;

exports[`Skeleton avatar element size 1`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-sm ant-skeleton-avatar-circle"
  />
</div>
`;

exports[`Skeleton avatar element size 2`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-circle"
  />
</div>
`;

exports[`Skeleton avatar element size 3`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
  />
</div>
`;

exports[`Skeleton avatar element size 4`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-avatar ant-skeleton-avatar-circle"
    style="width: 20px; height: 20px; line-height: 20px;"
  />
</div>
`;

exports[`Skeleton avatar shape 1`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar shape 2`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-square"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 1`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-sm ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 2`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 3`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 4`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-circle"
      style="width: 20px; height: 20px; line-height: 20px;"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton button element active 1`] = `
<div
  class="ant-skeleton ant-skeleton-element ant-skeleton-active"
>
  <span
    class="ant-skeleton-button"
  />
</div>
`;

exports[`Skeleton button element shape 1`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button"
  />
</div>
`;

exports[`Skeleton button element shape 2`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button ant-skeleton-button-round"
  />
</div>
`;

exports[`Skeleton button element shape 3`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button ant-skeleton-button-circle"
  />
</div>
`;

exports[`Skeleton button element size 1`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button"
  />
</div>
`;

exports[`Skeleton button element size 2`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button ant-skeleton-button-lg"
  />
</div>
`;

exports[`Skeleton button element size 3`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-button ant-skeleton-button-sm"
  />
</div>
`;

exports[`Skeleton input element active 1`] = `
<div
  class="ant-skeleton ant-skeleton-element ant-skeleton-active"
>
  <span
    class="ant-skeleton-input"
  />
</div>
`;

exports[`Skeleton input element size 1`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-input ant-skeleton-input-sm"
  />
</div>
`;

exports[`Skeleton input element size 2`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-input"
  />
</div>
`;

exports[`Skeleton input element size 3`] = `
<div
  class="ant-skeleton ant-skeleton-element"
>
  <span
    class="ant-skeleton-input ant-skeleton-input-lg"
  />
</div>
`;

exports[`Skeleton paragraph rows 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li />
      <li />
      <li
        style="width: 61%;"
      />
    </ul>
  </div>
</div>
`;

exports[`Skeleton paragraph width 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width: 93%;"
      />
    </ul>
  </div>
</div>
`;

exports[`Skeleton paragraph width 2`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li
        style="width: 28%;"
      />
      <li
        style="width: 93%;"
      />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-skeleton ant-skeleton-rtl"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width:38%"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width:61%"
      />
    </ul>
  </div>
</div>
`;

exports[`Skeleton should square avatar 1`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-square"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
    />
  </div>
</div>
`;

exports[`Skeleton should without avatar and paragraph 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
    />
  </div>
</div>
`;

exports[`Skeleton title width 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 93%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width: 61%;"
      />
    </ul>
  </div>
</div>
`;
